ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -12,204 +12,188 @@
+      Number.isNaN(Number.parseFloat(point[1])))
+  ) {
+    return true
+  } else if (
+    point[0] > 180 ||
+    point[0] < -180 ||
+    point[1] > 90 ||
+    point[1] < -90
+  ) {
+    return true
   }
-  validatePoint(point) {
+  return false
+}
+
+const BaseLine = props => {
+  const { label, geometryKey, setState, unitKey, widthKey, mode } = props
+  const [currentValue, setCurrentValue] = useState(
+    JSON.stringify(props[geometryKey])
+  )
+  const [baseLineError, setBaseLineError] = useState({
+    error: false,
+    message: '',
+  })
+  const [bufferError, setBufferError] = useState({ error: false, message: '' })
+
+  useEffect(
+    () => {
+      const { geometryKey } = props
+      setCurrentValue(JSON.stringify(props[geometryKey]))
+    },
+    [props.polygon, props.line]
+  )
+
+  function validateListOfPoints(coordinates) {
+    let message = ''
+    const isLine = mode.includes('line')
+    const numPoints = isLine ? 2 : 4
     if (
-      point.length !== 2 ||
-      (Number.isNaN(Number.parseFloat(point[0])) &&
-        Number.isNaN(Number.parseFloat(point[1])))
-    ) {
-      return JSON.stringify(point) + ' is not a valid point.'
-    } else if (
-      point[0] > 180 ||
-      point[0] < -180 ||
-      point[1] > 90 ||
-      point[1] < -90
+      !mode.includes('multi') &&
+      !coordinates.some(coords => coords.length > 2) &&
+      coordinates.length < numPoints
     ) {
-      return JSON.stringify(point) + ' is not a valid point.'
+      message = `Minimum of ${numPoints} points needed for ${
+        isLine ? 'Line' : 'Polygon'
+      }`
     }
-    return ''
-  }
-  validateListOfPoints(coordinates) {
-    let message = ''
-    const isLine = this.props.mode.includes('line')
-    let numPoints = isLine ? 2 : 4
-    if (!this.props.mode.includes('multi')) {
-      if (coordinates.some(coords => coords.length > 2)) {
-        message = ''
-      } else if (coordinates.length < numPoints) {
-        message = `Minimum of ${numPoints} points needed for ${
-          isLine ? 'Line' : 'Polygon'
-        }`
-      }
-    }
-    for (let i = 0; i < coordinates.length; i++) {
-      if (coordinates[i].length > 2) {
-        coordinates[i].forEach(coordinate => {
-          if (this.validatePoint(coordinate)) {
-            message = this.validatePoint(coordinate)
-          }
+    coordinates.forEach(coordinate => {
+      if (coordinate.length > 2) {
+        coordinate.forEach(coord => {
+          if (hasPointError(coord))
+          message = JSON.stringify(coord) + ' is not a valid point.'
         })
       } else {
-        if (this.props.mode.includes('multi')) {
+        if (mode.includes('multi')) {
           message = `Switch to ${isLine ? 'Line' : 'Polygon'}`
-        } else if (this.validatePoint(coordinates[i])) {
-          message = this.validatePoint(coordinates[i])
+        } else if (hasPointError(coordinate)) {
+          message = JSON.stringify(coordinate) + ' is not a valid point.'
```

Comment:
coordinate[i]

Reply:
wait i think `coordinate[i]` is the same as `coordinate` in this case because of changing the for loop to a for each